erpnext/erpnext_integrations/doctype/taxjar_settings/taxjar_settings.py

```diff
@@ -4,9 +4,98 @@
 
 from __future__ import unicode_literals
 
-# import frappe
+import json
+import os
+
+import frappe
+from frappe.custom.doctype.custom_field.custom_field import create_custom_fields
 from frappe.model.document import Document
+from frappe.permissions import add_permission, update_permission_property
+
+from erpnext.erpnext_integrations.taxjar_integration import get_client
 
 
 class TaxJarSettings(Document):
-	pass
+
+	def on_update(self):
+		TAXJAR_CREATE_TRANSACTIONS = frappe.db.get_single_value("TaxJar Settings", "taxjar_create_transactions")
+		TAXJAR_CALCULATE_TAX = frappe.db.get_single_value("TaxJar Settings", "taxjar_calculate_tax")
+		TAXJAR_SANDBOX_MODE = frappe.db.get_single_value("TaxJar Settings", "is_sandbox")
+		fields_hidden = 0
+
+		custom_fields = []
+		for dt in ['Item', 'Sales Invoice Item']:
+			doc = frappe.get_doc('Custom Field', {'dt': dt, 'fieldname':'product_tax_category'})
```

Comment:
`get_doc` will fail and stop execution if it cannot find the custom field. You should probably do a `frappe.db.exists` call to check for the field, and a separate `frappe.db.get_value` call to retrieve the value for `hidden`. That'll still be lighter than a `get_doc` call.